Fix: Break Manager

beanbag44 opened the pull request:
This introduces a more immutable style system for item swapping logic. Each tick a new SwapInfo is created for all active break infos.

This also fixes an issue with redundancy spam due to the build tasks request logic. 

Along with the fixes, ive added a new "Server Swap" setting for the user to edit how many ticks the server should be given to recognise the selected tool. This is hard limited to a minimum of 2 ticks for secondary breaks to allow the user to select 0 for servers < 1.21.5 that dont apply certain enchantments, like efficiency, as player attributes

Avanatiker commented:
Thanks a lot! will check it out later

Avanatiker commented:
Sry for not merging yet. I tried to make it more stable while realizing that our current plugin menu should get a big overhaul and get moved to the ClickGUI so people find the plugin stuff. That will take some time.

scorbett123 commented:
> Sry for not merging yet. I tried to make it more stable while realizing that our current plugin menu should get a big overhaul and get moved to the ClickGUI so people find the plugin stuff. That will take some time.

What issues have you been having with it? I have been testing it so far with no issues (except the running out of github api requests).

Avanatiker commented:
The problem is right now i need to fix these first:
- updating to Kotlin 1.5.30 causes some weird bugs for the PluginConfig class
- problem with loading hud settings (may be connected with first issue)
- move plugin list from seperate GUI to ClickGUI
- solve the API request limit problem

ToxicAven commented:
bump

Avanatiker commented:
Ok i finally did the plugin gui refactor and implemented your costum class loader into it. Sadly this pr is not working on the new code base bc it also contains code for plugin updating using the old manager. I have to close this but thanks for contributing the code for the loader and updater!

beanbag44 committed:
move almost all swap data and logic into the SwapInfo class, and add new serverSwap setting.

this seems to be on par with 1.20.4 consistency but there are improvements to be made regarding  speed

beanbag44 committed:
min server swap ticks to 2 for secondary breaks allowing for faster primary breaks as to not drag them down

beanbag44 committed:
Merge branch '1.21.5' into improvement/breakmanager-again

beanbag44 committed:
idk what i was thinking at 5 am

beanbag44 committed:
catch rebreak logic up to the rest and raise minimum secondary swap ticks to 3

beanbag44 committed:
Merge branch '1.21.5' into improvement/breakmanager-again

beanbag44 committed:
Merge branch 'refs/heads/1.21.5' into improvement/breakmanager-again

beanbag44 committed:
Merge branch 'refs/heads/1.21.5' into improvement/breakmanager-again

beanbag44 committed:
omit population from the repeat loop

beanbag44 committed:
pass all contexts over to the break manager for it to decide which it can take for optimal efficiency and leave stack selection check to the break manager too

beanbag44 committed:
min secondary swap ticks to 2

beanbag44 committed:
back to two...
